### DISCUSSION TITLE
Schematron and Java17

### DISCUSSION THREAD

**Discussion by Mattonius5** [phax/ph-schematron#167] [General]
Hello Philip,
I wonder how to proceed with Java17. Up until now, I've been using Spring Boot 2.7.x with Java 8.
I want to upgrade system to run on Java 17 and Spring Boot 3.2.X.
In J8 we were using JavaX libraries to marshall/unmarshall XML documents / java classes.
Spring Boot 3.2 with Java 17 is no longer compatible with old javax.xml.bind libraries thus I refactored everything to jakarta.xml.bind, Marshalling and unmarshalling works without problem using Spring dependent versions of glassfish.jaxb/jakarta.xml.bind.

Problem comes when I want to validate xml file using schematron (ph-schematron:5.6.5). Under Java 8 everything worked like charm.
Whenever I try to run "SchematronResourceSCH.fromClassPath("schematron/schemRules.sch")" I recieve and error of "java.lang.ClassNotFoundException: **javax.xml.bind**.JAXBElement".

I suppose this comes from schematron being only java8 compatible? How am I able to validate against Schematron using java17?
Thank you for your help,
Matt

**Comment by Mattonius5** [comment]
It turned out if I use "com.helger.schematron:ph-schematron-xslt:8.0.0" then it seems to be working, However I am not sure if this one should be used as I cannot find any compatibility matrix for individual artifacts and respective java versions.

**Reply by phax** [reply]
Hi @Mattonius5  please see https://github.com/phax/ph-schematron/wiki/News-and-noteworthy  for the listing of changes.
That e.g. shows that v7 was the step to Java 11 etc.
In general I recommend to always stay with the latest and greates...
